### PR TITLE
adds dataset generation files

### PR DESCRIPTION
**Pull request by adiebold**
parse textbook, turn parsed textbook into question&answer pairs, re-parse responses that failed to get parsed to JSON (must manually fix issues in the txt files before this step)

### CODE REVIEW

File: dataset_generation/__init__.py
Comment (81reap) [inline dataset_generation/__init__.py:1]: nit :: I don't think we need this file as we are not defining a package, just some scripts

File: dataset_generation/parse_textbook.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+from argparse import ArgumentParser
+import json
+import os
+import fitz # if missing, pip install pymupdf
```
Comment (81reap) [inline dataset_generation/parse_textbook.py:4]: nit :: could you add a requirements.txt for this? or include a list of all the dependencies you need on the `./dataset_generation/README.md` file?

**Review by 81reap — APPROVED** [review]
Consider adding a `./dataset_generation/README.md` as to how to use these scripts and files. It will help to recreate if we ever need to do it.

File: dataset_generation/__init__.py
Comment (adiebold) [reply dataset_generation/__init__.py:1]: don't we need this in order to be able to import from config.py?

File: dataset_generation/parse_textbook.py (hunk 1 above)
Comment (adiebold) [reply dataset_generation/parse_textbook.py:4]: will do, I'll make another PR with a README and requirements.txt

**Pull request merged**